plugins/services/tasks/local.go

```diff
@@ -395,15 +396,25 @@ func (l *local) List(ctx context.Context, r *api.ListTasksRequest, _ ...grpc.Cal
 }
 
 func addTasks(ctx context.Context, r *api.ListTasksResponse, tasks []runtime.Task) {
+	var wg sync.WaitGroup
+	taskChan := make(chan *task.Process, len(tasks))
 	for _, t := range tasks {
-		tt, err := getProcessState(ctx, t)
-		if err != nil {
-			if !errdefs.IsNotFound(err) { // handle race with deletion
-				log.G(ctx).WithError(err).WithField("id", t.ID()).Error("converting task to protobuf")
+		wg.Add(1)
+		go func(ctx context.Context, t runtime.Process) {
```

Comment:
We may want to limit the total number of goroutines rather than having one goroutine per task (and no bound on the number of goroutines).  I'm worried about the total amount of CPU that containerd might use to list all tasks on nodes with many tasks.

Reply:
these goroutines will split by calling different shims ttrpc server, which limit do you want to set ? @samuelkarp